tests/src/test/java/org/wso2/micro/gateway/tests/common/BaseTestCase.java

```diff
@@ -44,46 +45,40 @@
 
     protected void init(String label, String project, String security) throws Exception {
         CLIExecutor cliExecutor;
-
-        microGWServer = ServerInstance.initMicroGwServer();
-        String cliHome = microGWServer.getServerHome();
+        String configPath = Objects.requireNonNull(getClass().getClassLoader()
+                .getResource("confs" + File.separator + "default-test-config.conf")).getPath();
+        microGWServer = ServerInstance.initMicroGwServer(configPath);
+        String cliHome = microGWServer.getToolkitDir();
 
         boolean isOpen = Utils.isPortOpen(MOCK_SERVER_PORT);
         Assert.assertFalse(isOpen, "Port: " + MOCK_SERVER_PORT + " already in use.");
         mockHttpServer = new MockHttpServer(MOCK_SERVER_PORT);
         mockHttpServer.start();
-        //System.setProperty(GatewayCliConstants.SYS_PROP_SECURITY, "oauth2");
         cliExecutor = CLIExecutor.getInstance();
         cliExecutor.setCliHome(cliHome);
-        cliExecutor.generate(label, project, security);
+        cliExecutor.generate(label, project);
 
         String balPath = CLIExecutor.getInstance().getLabelBalx(project);
-        String configPath = getClass().getClassLoader()
-                .getResource("confs" + File.separator + "default-test-config.conf").getPath();
-        String[] args = {"--config", configPath, "--experimental"};
-        microGWServer.startMicroGwServer(balPath, args);
+        microGWServer.startMicroGwServer(balPath);
     }
 
     protected void init(String project, String[] openAPIFileNames) throws Exception {
         CLIExecutor cliExecutor;
-
-        microGWServer = ServerInstance.initMicroGwServer();
-        String cliHome = microGWServer.getServerHome();
+        String configPath = Objects.requireNonNull(getClass().getClassLoader()
+                .getResource("confs" + File.separator + "default-test-config.conf")).getPath();
+        microGWServer = ServerInstance.initMicroGwServer(configPath);
+        String cliHome = microGWServer.getToolkitDir();
 
         boolean isOpen = Utils.isPortOpen(MOCK_SERVER_PORT);
         Assert.assertFalse(isOpen, "Port: " + MOCK_SERVER_PORT + " already in use.");
         mockHttpServer = new MockHttpServer(MOCK_SERVER_PORT);
         mockHttpServer.start();
-        //System.setProperty(GatewayCliConstants.SYS_PROP_SECURITY, "oauth2");
         cliExecutor = CLIExecutor.getInstance();
         cliExecutor.setCliHome(cliHome);
         cliExecutor.generateFromDefinition(project, openAPIFileNames);
 
         String balPath = CLIExecutor.getInstance().getLabelBalx(project);
-        String configPath = getClass().getClassLoader()
-                .getResource("confs" + File.separator + "default-test-config.conf").getPath();
-        String[] args = {"--config", configPath, "--experimental"};
-        microGWServer.startMicroGwServer(balPath, args);
+        microGWServer.startMicroGwServer(balPath);
```

Comment:
We need a way to pass arguments to server startup command

Reply:
There is an existing overloaded method for that purpose. It can be reused.

Reply:
added with https://github.com/wso2/product-microgateway/pull/641/commits/901bd5a0e547ce5bf8f6e7b586c2faf92197121d.